authui/src/authflowv2/botprotection/cloudflareTurnstileError.ts

```diff
@@ -0,0 +1,95 @@
+// These are meaningful errors types for handling, which originated from actual cloudflare turnstile error codes
+export enum CloudflareTurnstileError {
+  InvalidSiteKey = "invalid-sitekey",
+  DomainNotAllowed = "domain-not-allowed",
+  UnusualVisitorBehavior = "unusual-visitor-behavior",
+  UnsupportedBrowser = "unsupported-browser",
+  Timeout = "timeout",
+  Unknown = "unknown",
+}
+
+export interface CloudflareTurnstileErrorCodeParseResult {
+  error: CloudflareTurnstileError;
+  shouldRetry: boolean;
+  shouldDisplayErrMsg: boolean;
+}
+
+const CLOUDFLARE_TURNSTILE_ERROR_INVALID_SITE_KEY: CloudflareTurnstileErrorCodeParseResult =
+  {
+    error: CloudflareTurnstileError.InvalidSiteKey,
+    shouldRetry: false,
+    shouldDisplayErrMsg: false, // cloudflare will render the error message inside its widget
+  };
+
+const CLOUDFLARE_TURNSTILE_ERROR_DOMAIN_NOT_ALLOWED: CloudflareTurnstileErrorCodeParseResult =
+  {
+    error: CloudflareTurnstileError.DomainNotAllowed,
+    shouldRetry: false,
+    shouldDisplayErrMsg: false, // cloudflare will render the error message inside its widget
+  };
+const CLOUDFLARE_TURNSTILE_ERROR_UNUSUAL_VISITOR_BEHAVIOR: CloudflareTurnstileErrorCodeParseResult =
+  {
+    error: CloudflareTurnstileError.UnusualVisitorBehavior,
+    shouldRetry: true,
+    shouldDisplayErrMsg: false,
+  };
+const CLOUDFLARE_TURNSTILE_ERROR_UNSUPPORTED_BROWSER: CloudflareTurnstileErrorCodeParseResult =
+  {
+    error: CloudflareTurnstileError.UnsupportedBrowser,
+    shouldRetry: false,
+    shouldDisplayErrMsg: false, // cloudflare will render the error message inside its widget
+  };
+const CLOUDFLARE_TURNSTILE_ERROR_TIMEOUT: CloudflareTurnstileErrorCodeParseResult =
+  {
+    error: CloudflareTurnstileError.Timeout,
+    shouldRetry: true,
+    shouldDisplayErrMsg: false, // auto retry, no need err message
+  };
+const CLOUDFLARE_TURNSTILE_ERROR_UNKNOWN: CloudflareTurnstileErrorCodeParseResult =
+  {
+    error: CloudflareTurnstileError.Unknown,
+    shouldRetry: false,
+    shouldDisplayErrMsg: true,
+  };
+
+const CLOUDFLARE_TURNSTILE_ERROR_CODE_REGEX = /\d{6}/;
+
+export function isCloudflareTurnstileErrorCodeValid(errCode: string): boolean {
+  return CLOUDFLARE_TURNSTILE_ERROR_CODE_REGEX.test(errCode);
+}
+
+/**
+ * @see https://developers.cloudflare.com/turnstile/troubleshooting/client-side-errors/error-codes/
+ */
+export function parseCloudflareTurnstileErrorCode(
+  errCode: string
+): CloudflareTurnstileErrorCodeParseResult {
+  if (!isCloudflareTurnstileErrorCodeValid(errCode)) {
+    throw new Error(`unrecognised cloudflare turnstile error code ${errCode}`);
+  }
```

Comment:
No need to validate the error code. We just need to handle the documented error code, and handle unknown error code by logging and showing alert-message.